python/amici/ode_export.py

```diff
@@ -2103,6 +2109,48 @@ def _writeFunctionFile(self, function):
         ) as fileout:
             fileout.write('\n'.join(lines))
 
+    def _write_function_index(self, function, indextype):
+        """Generate equations and write the C++ code for the function
+        `function`.
+
+        Arguments:
+            function: name of the function to be written (see self.functions)
+            @type str
+            indextype: type of index {'colptrs', 'rowvals'}
+
+        Returns:
+
+        Raises:
+
+        """
+
+        # function signature
+        signature = f'(sunindextype *{indextype})'
+
+        if indextype == 'colptrs':
+            values = self.model.colptrs(function)
+        elif indextype == 'rowvals':
+            values = self.model.rowvals(function)
+        else:
+            raise ValueError('Invalid value for type, must be colptr or '
+                             'rowval')
+
+        lines = list()
+        lines.append('#include "sundials/sundials_types.h"')
```

Comment:
Combining `append`s would avoid some reallocations. Not sure if measurable though.